R/negloglik.laplace.R

```diff
@@ -1,16 +1,20 @@
 "negloglik.laplace" <-
-function(xpar, xx, ss)
+function(xpar, xx, ss, tlo, thi)
 {
 #
 #  marginal negative log likelihood function for laplace prior
 #  
 #  xx data
 #  xpar vector of two parameters:
-#      xpar[1] :  threshold of minimum sd
+#      xpar[1] :  w standardized to [0, 1] given a
 #      xpar[2] :  scale factor a
 #
 	a <- xpar[2]
-	w <- wfromt(xpar[1], min(ss), a = a)
-	loglik <- sum(log(1 + w * matrix(beta.laplace(xx, ss, a), nrow=1)))
+	# Calculate the range of w given a, using negative monotonicity between w and t
+	wlo <- wfromt(thi, ss, a = a)
+	whi <- wfromt(tlo, ss, a = a)
+	wlo <- max(wlo)
+	whi <- min(whi)
+	loglik <- sum(log(1 + (xpar[1] * (whi - wlo) + wlo) * beta.laplace(xx, ss, a)))
```

Comment:
Please refer to Equation 2.21 in the report. wlo and whi is the lower and upper bound of weight. wlo is actually 1 here so it's not necessary to write it in this way.